contracts/Pool.sol

```diff
@@ -1,8 +1,35 @@
 // SPDX-License-Identifier: GPL-3.0-only
 pragma solidity ^0.6.2;
 
-/// @notice Funding pool contract.
-/// Allows sending ETH automatically to a configurable set of receivers.
+/// @notice Funding pool contract. Automatically sends funds to a configurable set of receivers.
+///
+/// The contract has 2 types of users: the senders and the receivers.
+///
+/// A sender has some funds and a set of addresses of receivers, to whom he wants to send funds.
+/// In order to send there are 3 conditions, which must be fulfilled:
+///
+/// 1. There must be funds on his account in this contract.
+///    They can be added with `topUp` and removed with `withdraw`.
+/// 2. Total amount sent to the receivers on each block must be set.
+///    This is done with `setAmountPerBlock`.
+/// 3. A set of receivers must be non-empty.
+///    Receivers can be added, removed and updated with `setReceiver`.
+///    Each receiver has a weight, which is used to calculate how the total sent amount is split.
+///
+/// Each of these functions can be called in any order and at any time, they have immediate effects.
+/// When all of these conditions are fulfilled, on each block the configured amount is being sent.
+/// It's extracted from the `withdraw`able balance and transferred to the receivers.
+/// The process continues automatically until the sender's balance is empty.
```

Comment:
I think it's confusing to explain this as "on each block the configured amount is being sent", at the contract level (this sounds more like an explanation for end users). We should be precise here and say that on *on each block, the amount of funds available to collect by the receivers increases*.

Reply:
I think that it's the right abstraction to explain what the contract is doing and how it should be used. A low-level description expressed in recalculations being enabled based on time passage would needlessly obscure the view. I don't think that anybody except us really cares either, it's just an implementation detail.

I've added a paragraph explaining that things "happening periodically" is just a facade. I think that it should be enough to un-confuse people wondering how is it supposed to work. What do you think?

Reply:
Sounds good!